### PR TITLE
feat(Integrationtest): integration test param support

### PR DESCRIPTION
**Pull request by abhinandan13jan**
## Fixes 

https://issues.redhat.com/browse/HAC-4596

## Description

- Adds parameter support for IntegrationTest
- Creates a Param component

## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Screen shots / Gifs for design review 

Screenshot 2023-09-13 at 9 26 52 PM
Screenshot 2023-09-13 at 9 52 22 PM
Screenshot 2023-09-13 at 9 54 32 PM
Screenshot 2023-09-13 at 9 54 43 PM
Screenshot 2023-09-13 at 9 54 57 PM

## Browser conformance: 

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Comment by MariaLeonova** [comment]
@abhinandan13jan this looks great! I just have one question: why is there a grey outline around the parameters field?

**Comment by Misjohns** [comment]
There's a few elements that look off. I've included a redline with recommendations. Please refer to this demo for padding and text styling: https://www.patternfly.org/components/forms/form
image

**Comment by abhinandan13jan** [comment]
@MariaLeonova By default, the expandable card has an outline, removed it.

Screenshot 2023-09-13 at 6 28 53 PM

@Misjohns 
1. Updated the spacing. Was previously using the default spacing of patternfly Data-List
2. Removed the bold weight from `Parameter 1` name
3. The `Parameters` font size is exactly the same as other fields in the `IntegrationTestForm`. If we make it larger will look different from other fields like `name` `git url`. Do let me know if it needs to be updated

Screenshot 2023-09-13 at 7 09 20 PM

**Comment by Misjohns** [comment]
The expandable section **Parameters** isn't a field label so it should be same size as the "Mark as optional..."
https://www.patternfly.org/components/expandable-section

**Comment by abhinandan13jan** [comment]
> The expandable section **Parameters** isn't a field label so it should be same size as the "Mark as optional..." https://www.patternfly.org/components/expandable-section

@Misjohns updated
Screenshot 2023-09-13 at 9 52 22 PM
Screenshot 2023-09-13 at 9 54 32 PM
Screenshot 2023-09-13 at 9 54 57 PM

**Comment by Misjohns** [comment]
LGTM!

File: src/components/IntegrationTest/IntegrationTestParams.tsx [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+    }
+    if (Array.isArray(parameters[i].values) && parameters[i].values.length === 1) {
+      currentParams[i].value = currentParams[i].values[0];
+      currentParams[i].values = null;
+      setValue(currentParams);
+      return;
+    }
+  };
+
+  const toggleExpandedState = (i) => {
```
Comment (rohitkrai03) [inline src/components/IntegrationTest/IntegrationTestParams.tsx:117]: Isn't there a uncontrolled version of expand collapse section which handles the state automatically?

File: src/components/IntegrationTest/IntegrationTestParams.tsx [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+          className="pf-v5-u-pl-xs pf-v5-u-pr-xs"
+        >
+          {heading ?? 'Parameters'}
+          {error && (
+            <Alert data-test="its-param-error-alert" title={error} variant={AlertVariant.danger} />
+          )}
+        </CardHeader>
+        <CardExpandableContent>
+          <CardBody className="review-component-card__card-body">
+            {Array.isArray(parameters) && parameters.length > 0 ? (
```
Comment (rohitkrai03) [inline src/components/IntegrationTest/IntegrationTestParams.tsx:151]: Wouldn't it be better and easier to handle this using `FieldArray` by Formik?

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]

File: src/components/IntegrationTest/IntegrationTestParams.tsx (hunk 1 above)
Comment (abhinandan13jan) [reply src/components/IntegrationTest/IntegrationTestParams.tsx:53]: I am not sure, but I also need to expand new params when created.

File: src/components/IntegrationTest/FormikParamsField.tsx [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+                            >
+                              <DataListItemRow className="pf-v5-u-pl-0">
+                                <DataListItemCells
+                                  dataListCells={[
+                                    <DataListCell
+                                      key="param-name"
+                                      width={3}
+                                      className="pf-v5-u-pl-xl pf-v5-u-pt-0"
+                                    >
+                                      <FormGroup label="Name" isRequired>
```
Comment (karthikjeeyar) [inline src/components/IntegrationTest/FormikParamsField.tsx:158]: Name and value field is optional in design doc.

```suggestion
                                      <FormGroup label="Name">
```
image

**Review by karthikjeeyar — COMMENTED** [review]
While creating the integration test we should not set the param value if it is empty. Right now if you click on add value multiple times and create a integration test then it sets empty values in spec.params as you can see here below in the Edit integration test flow.

editIntegraionTest

**Comment by karthikjeeyar** [comment]
Missing Help tooltip which is outlined in the design

 design doc:
Screenshot 2023-09-27 at 1 53 22 PM

**Comment by abhinandan13jan** [comment]
> Missing Help tooltip which is outlined in the design
> 
> design doc: Screenshot 2023-09-27 at 1 53 22 PM

So we decided in the meeting, not to use a tooltip, as this is a straightforward field and we dont have tooltips for any other fields.
